airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/SchedulerHandler.java

```diff
@@ -393,10 +411,13 @@ public JobInfoRead cancelJob(final JobIdRequestBody jobIdRequestBody) throws IOE
   // determine whether 1. the source schema change resulted in a broken connection or 2. the user
   // wants the connection disabled when non-breaking changes are detected. If so, disable that
   // connection. Modify the current discoveredSchema object to add a CatalogDiff,
-  // containsBreakingChange paramter, and connectionStatus parameter.
+  // containsBreakingChange parameter, and connectionStatus parameter.
   private void generateCatalogDiffsAndDisableConnectionsIfNeeded(final SourceDiscoverSchemaRead discoveredSchema,
-                                                                 final SourceDiscoverSchemaRequestBody discoverSchemaRequestBody)
+                                                                 final SourceDiscoverSchemaRequestBody discoverSchemaRequestBody,
+                                                                 final UUID workspaceId)
       throws JsonValidationException, ConfigNotFoundException, IOException {
+    final ConnectionNotificationWorkflow notificationWorkflow =
```

Comment:
So far, I think the temporal access have been generally abstracted ([ConnectionManagerUtils](https://github.com/airbytehq/airbyte/blob/3c52168f80cdc14cf346b4cadf5660fff0ad4936/airbyte-commons-temporal/src/main/java/io/airbyte/commons/temporal/ConnectionManagerUtils.java)), should we either extend this object or have a similar abstraction for the notifications (NotificationClient could be a fit)?

Reply:
Yes I agree, the scheduler shouldn't be aware of temporal. They should launch the temporal action through an interface/utils that could be replace by something else than temporal